cpp/order/HexaticTranslational.h

```diff
@@ -26,10 +26,10 @@ template<typename T> class HexaticTranslational
 {
 public:
     //! Constructor
-    HexaticTranslational(T k, bool weighted = false) : m_k(k), m_weighted(weighted) {}
+    explicit HexaticTranslational(T k, bool weighted = false) : m_k(k), m_weighted(weighted) {}
```

Comment:
This is more of a cpp question, but what does explicit do to a contructor?

Reply:
`explicit` is a good thing to add to one-argument constructors because it prevents unintended typecasting. (The second argument has a default value so it's like having a one argument constructor.)

https://en.cppreference.com/w/cpp/language/explicit
https://www.reddit.com/r/Cplusplus/comments/ejuf7k/must_i_use_explicit_keyword_in_c/

Reply:
Yeah you don't want to be able to do something like `Hexatic hex = 6`, which would otherwise be possible.

Reply:
This seems like something that should just always happen implicitly for safety reasons. I wouldn't want to have 2, 3, 4, ... argument constructors doing what Vyas showed above either.

Reply:
While it is true that the same principle applies to (n>1)-argument constructors, in practice you're not very likely to accidentally do something like `Hexatic hex = {6, true}`, so I don't think linters check for that. We could go through and mark all constructors this way, but it seems like overkill.

Reply:
> This seems like something that should just always happen implicitly for safety reasons.
Did you mean _explicitly_? I think adding `explicit` is the right choice for single-argument constructors like this. As Vyas mentions, it's not really necessary for constructors with more than one argument.